File: sanghoon/programmers-2/큰 수 만들기.py
def solution(number, k):
    answer = []
    for num in number:
        while k > 0 and answer and answer[-1] < num:
            answer.pop()
            k -= 1
        answer.append(num)
    print(answer)
    print(answer[:len(answer) - k])
    print(''.join(answer[:len(answer) - k]))
    return answer

number = "4177252841"
k = 4
solution(number, k)

File: sanghyun/programmers-2/큰 수 만들기.java
import java.util.Stack;

class Solution {
    public String solution(String number, int k) {
        String answer = "";
        Stack<Integer> stack = new Stack<>();
        for (int i = 0; i < number.length(); i++) {
            int num = number.charAt(i) - '0';
            if (stack.isEmpty() || stack.peek() > num || k == 0) {
                stack.add(num);
            } else {
                while (stack.peek() < num){
                    stack.pop();
                    k--;
                    if(stack.isEmpty() || k == 0) break;
                }
                stack.add(num);
            }
        }
        for (int i = 0; i < stack.size() - k; i++) {
            answer += stack.get(i);
        }
        return answer;
    }
}
